lib/tern.js

```diff
@@ -660,9 +660,14 @@
     var typeInfo = query.types || query.docs || query.urls || query.origins;
     var wrapAsObjs = typeInfo || query.depths;
 
-    for (var i = 0; i < completions.length; ++i) {
+    for (var i = 0, len = completions.length; i < len; i++) {
       var c = completions[i];
-      if ((wrapAsObjs ? c.name : c) == name) return;
+      if ((wrapAsObjs ? c.name : c) == name) {
+        if(query.origins && aval && c.origin !== aval.origin) {
```

Comment:
Why not just add the inverse of this condition to the `if` above, rather than introducing extra control flow?